frame/instrumental-strategy-pablo/src/lib.rs

```diff
@@ -294,18 +364,63 @@ pub mod pallet {
 						pool_id_and_state.state == State::Normal,
 						Error::<T>::TransferringInProgress
 					);
-					Pools::<T>::mutate(asset_id, |_| PoolState { pool_id, state: State::Normal });
+					Pools::<T>::mutate(asset_id, |pool| {
+						*pool = Some(PoolState { pool_id, state: State::Transferring });
+					});
+					Self::deposit_event(Event::OccuredFundsTransferring {
+						old_pool_id: pool_id_and_state.pool_id,
+						new_pool_id: pool_id,
+					});
+					Self::transferring_funds_from_old_pool_to_new(
+						asset_id,
+						pool_id_and_state.pool_id,
+						pool_id,
+					)?;
 				},
 				Err(_) => Pools::<T>::insert(asset_id, PoolState { pool_id, state: State::Normal }),
 			}
 			Ok(())
 		}
+		/// Called when an existing asset_id-pool_id pair is replaced by new pool_id.
+		/// Occurs a withdrawal of all funds from the old pool to the Vault, and then from the Vault
+		/// to the new pool.
+		#[transactional]
+		fn transferring_funds_from_old_pool_to_new(
```

Comment:
This function, and the other Pablo-specific functions, shouldn't be apart of the `InstrumentalProtocolStrategy` trait - you can create an `InstrumentalPabloStrategy` trait to house the logic specific to this pallet.

Reply:
the `InstrumentalProtocolStrategy` should only contain the functions that all Instrumental Strategies will have to implement.

Reply:
Thinking about it you should even name `InstrumentalProtocolStrategy` to just `InstrumentalStrategy` as assuming they will only speak to one protocol was naive of me.